### DISCUSSION TITLE
How to use values.yaml in flux deployment

### DISCUSSION THREAD

**Discussion by SonalinP** [fluxcd/flux2#4089] [General]
Hi, 

I have different  values.yaml for different  environment for one application helm chart.
How can I deploy with different values.yaml via flux ? We can append the values in helmrelease but i want to use the existing values.yaml in the deployment via flux . 
Where in runtime values.yaml should be appended and deployed

**Comment by makkes** [comment]
You can use multiple sources for the values of a HelmRelease: https://fluxcd.io/flux/components/helm/helmreleases/#values-overrides

If you want to maintain your `values.yaml` files in the Git repo rather than ConfigMaps, you can use kustomize's `configMapGenerator`: https://fluxcd.io/flux/guides/helmreleases/#refer-to-values-in-configmaps-generated-with-kustomize

**Reply by SonalinP** [reply]
Thanks Makkes.
I tried the above method . 
helm upgrade is failing with below :

**Warning  error   7m57s (x53 over 101m)   helm-controller  Helm upgrade failed: template: <service>/templates/deployment.yaml:34:27: executing "<service>/templates/deployment.yaml" at <.Values.env.CmUsr>: nil pointer evaluating interface {}.CmUsr**

I see the values present in config map generated by flux. This values.yaml is presenet in git repo along with kustomization.yaml and not part of helm chart . while the helm chart deployment is working fine if we deploy manually using "helm install <name> <chart-name> -f values.yaml.
 
deployment.yaml has values as below format .

        env:
  
        - name: CM_USER
          value: {{ .Values.env.cmUsr }}
        - name: CM_PASSWORD
          value: {{ .Values.env.cmPwd }}







Could you please suggest , what is wrong here ?
